auth_session_timeout/models/res_users.py

```diff
@@ -13,31 +14,110 @@
 from os.path import getmtime
 from time import time
 
+_logger = logging.getLogger(__name__)
+
 
 class ResUsers(models.Model):
     _inherit = 'res.users'
 
-    def _check_session_validity(self, db, uid, passwd):
-        if not request:
+    def _auth_timeout_ignoredurls_get(self):
+        """Pluggable method for calculating ignored urls
+        Defaults to stored config param
+        """
+        param_model = self.pool['ir.config_parameter']
+        return param_model._auth_timeout_get_parameter_ignoredurls()
+
+    def _auth_timeout_deadline_calculate(self):
+        """Pluggable method for calculating timeout deadline
+        Defaults to current time minus delay using delay stored as config param
+        """
+        param_model = self.pool['ir.config_parameter']
+        delay = param_model._auth_timeout_get_parameter_delay()
+        if delay is False or delay <= 0:
+            return False
+        return time() - delay
+
+    def _auth_timeout_session_terminate(self, session):
+        """Pluggable method for terminating a timed-out session
+
+        This is a late stage where a session timeout can be aborted.
+        Useful if you want to do some heavy checking, as it won't be
+        called unless the session inactivity deadline has been reached.
+
+        Return:
+            True: session terminated
+            False: session timeout cancelled
+        """
+        if session.db and session.uid:
+            session.logout(keep_db=True)
+        return True
+
+    def _auth_timeout_request_get(self):
+        return request
+
+    def _auth_timeout_session_filename_get(self, sid):
+        return root.session_store.get_session_filename(sid)
+
+    def _auth_timeout_utime(self, path, dates=None):
+        return utime(path, dates)
+
+    def _auth_timeout_getmtime(self, path):
+        return getmtime(path)
+
+    def _auth_timeout_check(self):
+        _request = self._auth_timeout_request_get()
+
+        if not _request:
+            return
+
+        session = _request.session
+
+        # Calculate deadline
+        deadline = self._auth_timeout_deadline_calculate()
+
+        # Check if past deadline
+        expired = False
+        if deadline is not False:
+            path = self._auth_timeout_session_filename_get(session.sid)
+            try:
+                expired = self._auth_timeout_getmtime(path) < deadline
+            except OSError as e:
+                _logger.warning(
+                    'Exception reading session file modified time: %s'
+                    % e
```

Comment:
I am using `warning` for these log entries.
Any reason I should use `debug` instead?